packages/openapi-fetch/src/index.d.ts

```diff
@@ -144,7 +144,11 @@ export function onRequest(
   req: MiddlewareRequest,
   options: MergedOptions,
 ): Request | undefined | Promise<Request | undefined>;
-export function onResponse(res: Response, options: MergedOptions): Response | undefined | Promise<Response | undefined>;
+export function onResponse(
+  res: Response,
+  options: MergedOptions,
+  req: MiddlewareRequest,
```

Comment:
👍 This is a good non-breaking change. Will probably opt  for a more “logical” ordering in a future breaking change